controllers/storagecluster/storageclient.go

```diff
@@ -97,10 +97,14 @@ func (s *storageClient) ensureDeleted(r *StorageClusterReconciler, storagecluste
 	}
 	storageClient := &ocsclientv1a1.StorageClient{}
 	storageClient.Name = storagecluster.Name
+	if err := r.Get(r.ctx, client.ObjectKeyFromObject(storageClient), storageClient); client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get storageclient %s: %v", storageClient.Name, err)
+	} else if storageClient.UID == "" {
+		return reconcile.Result{}, nil
+	}
 
-	if err := r.Delete(r.ctx, storageClient); err != nil && !kerrors.IsNotFound(err) {
-		r.Log.Error(err, "Failed to initiate deletion of local StorageClient CR")
-		return reconcile.Result{}, err
+	if err := r.Delete(r.ctx, storageClient); err != nil {
```

Comment:
Same here, Shouldn't the finalizer removal be done before Delete() is called on a object?